src/langcheck/metrics/en/reference_based_text_quality.py

```diff
@@ -4,14 +4,78 @@
 
 from rouge_score import rouge_scorer
 
-from langcheck.metrics._validation import validate_parameters_reference_based
+from langcheck.metrics._validation import (
+    validate_parameters_answer_correctness,
+    validate_parameters_reference_based,
+)
 from langcheck.metrics.eval_clients import EvalClient
 from langcheck.metrics.metric_value import MetricValue
 from langcheck.metrics.scorer.hf_models import (
     SentenceTransformerSimilarityScorer,
 )
 from langcheck.utils.progess_bar import tqdm_wrapper
 
+from ..prompts._utils import get_template
+
+
+def answer_correctness(
+    generated_outputs: List[str] | str,
+    reference_outputs: List[str] | str,
+    prompts: List[str] | str,
+    eval_model: EvalClient,
+) -> MetricValue[Optional[float]]:
+    """Calculates the correctness of the generated outputs. This metric takes on
+    float values of either 0.0 (Incorrect), 0.5 (Partially Correct), or 1.0
+    (Correct). The score may also be `None` if it could not be computed.
+
+    We currently only support the evaluation based on an EvalClient.
+
+    Args:
+        generated_outputs: The model generated output(s) to evaluate
+        reference_outputs: The reference output(s)
+        prompts: The prompts used to generate the output(s)
+        eval_model: The EvalClient instance used for the evaluation
+
+    Returns:
+        A :class:`~langcheck.metrics.metric_value.MetricValue` object
+    """
+    generated_outputs, reference_outputs, prompts = (
+        validate_parameters_answer_correctness(
+            generated_outputs, reference_outputs, prompts
+        )
+    )
+
+    answer_correctness_template = get_template(
```

Comment:
For other English metrics, the templates are loaded with `load_prompt_template` function now to use specific prompts for certain models (only used for Prometheus for now)

Given there is no Prometheus prompt I think it is okay to call `get_template` here for now, but we need to think when to write these model-specific prompts in the future

Reply:
Oh nice! I wrote this before `load_prompt_template` existed so I just missed it. Updated!

Reply:
Actually you need to write additional prompts for this to work. There is no `answer_correctness` prompt for Prometheus. Could you add a fallback mechanism to `load_prompt_template` so that it works properly?

Reply:
Ah yeah it would fail with a `FileNotFoundError`, which kind of makes sense but is not that helpful to the user I guess. I changed it to raise a ValueError that specifies that the metrics isn't supported by Prometheus
<img width="979" alt="image" src="https://github.com/citadel-ai/langcheck/assets/107823399/3d86c58d-36d6-4acb-b8c4-04c755427dd0">